items/scrolls.py

```diff
@@ -80,3 +80,32 @@ def action_activate(self, action: ActionWithItem) -> None:
             )
             actor.damage(self.damage)
         self.consume(action)
+
+
+class GenocideScroll(Scroll):
+    name = "Genocide Scroll"
+    color = (32, 32, 0)
+
+    def action_activate(self, action: ActionWithItem) -> None:
+        selected_xy = states.ingame.PickLocation(
+            action.model, "Select who to genecide.", action.actor.location.xy
+        ).loop()
+        if not selected_xy:
+            raise Impossible("Targeting canceled.")
+        if not action.map.visible.T[selected_xy]:
+            raise Impossible("You cannot target a enemy outside your field of view.")
+
+        selected_actor = action.map.fighter_at(*selected_xy)
+        if not selected_actor or selected_actor.is_player():
+            raise Impossible("No enemy selected to genocide.")
+
+        type_fighter = type(selected_actor.fighter)
+        # Use a copy of the actors list since it may be edited during the loop.
+        for actor in list(action.map.actors):
+            if isinstance(actor.fighter, type_fighter):
```

Comment:
`isinstance` might be too inclusive?  This engine doesn't really have proper categories for monsters.  This is probably fine.

Reply:
Hmm, maybe. But if the engine supported monster categories, I would like genocide concrete type of monster, not category